src/client/identity/identity.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package identity;
+option go_package = "github.com/pachyderm/pachyderm/src/client/identity";
+
+/* ConnectorConfig represents an IDP to delegate authentication to */
+message ConnectorConfig {
+  string id = 1;
+  string name = 2;
+  string type = 3;
+  int64 configVersion = 4;
+  string jsonConfig = 5;
```

Comment:
Conveniently Dex just treats IDP config as a JSON blob, like our existing API. This API should be able to configure any provider they support upstream with no changes.